target/arm/helper.c

```diff
@@ -9851,15 +11110,36 @@ static void arm_cpu_do_interrupt_aarch64(CPUState *cs)
         new_mode |= PSTATE_TCO;
     }
 
-    pstate_write(env, PSTATE_DAIF | new_mode);
+#ifdef TARGET_CHERI
+    env->pstate &= ~PSTATE_C64;
+    if (cap_exception && (env->CCTLR_el[new_el] & CCTRL_C64E))
+        env->pstate |= PSTATE_C64;
+#endif
```

Comment:
`pstate_write()` below will trash the C64 bit otherwise.
```suggestion
#ifdef TARGET_CHERI
    new_mode &= ~PSTATE_C64;
    if (cap_exception && (env->CCTLR_el[new_el] & CCTRL_C64E))
        new_mode |= PSTATE_C64;
#endif
```

Reply:
Oh oops... Something testing really should have found =/

Reply:
Yeah we should definitely report this as missing coverage.